service/waterfall.go

```diff
@@ -637,6 +638,23 @@ func (uis *UIServer) waterfallPage(w http.ResponseWriter, r *http.Request) {
 	}{newUILink, uis.Settings.Jira.Host, uis.GetCommonViewData(w, r, false, true)}, "base", "waterfall.html", "base_angular.html", "menu.html")
 }
 
+// Create and return a redirect to the spruce waterfall page if the user is opted in to the new UI.
+func (uis *UIServer) mainlineCommitsRedirect(w http.ResponseWriter, r *http.Request) {
+	u := gimlet.GetUser(r.Context())
+	if u != nil {
+		user, ok := u.(*user.DBUser)
+		if ok {
+			if user.Settings.UseSpruceOptions.SpruceV1 {
+				evergreen.GetEnvironment().Settings().Ui.UIv2Url = ""
+				http.Redirect(w, r, fmt.Sprintf("%s/commits/", uis.Settings.Ui.UIv2Url), http.StatusSeeOther)
+			}
+		}
+	}
+
+	http.Redirect(w, r, "/waterfall", http.StatusSeeOther)
```

Comment:
This redirects `/` to `/waterfall`. How does `/waterfall` redirect to spruce?

Reply:
Oh sorry, I misread, resolving.

Reply:
I added a comment to make it more clear. This is essentially a fallback. If there is no user in the request/ or they do not have spruce enabled. We fallback and redirect to "/waterfall"